src/foam/dao/index/LRUCachingIndex.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao.index;
+
+public class LRUCachingIndex
+    extends ProxyIndex
+{
+  public static long
+
+  protected int size_ = 0;
+  protected final int maxSize;
```

Comment:
Rename maxSize_. We follow the convention of have protected instance variables end in _.